projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/options/LLVMBaseOption.java

```diff
@@ -29,131 +29,61 @@
-    TRACE_EXECUTION("TraceExecution", "Trace execution, printing each SSA assignment", false, LLVMOptions::parseBoolean, PropertyCategory.DEBUG);
-
-    LLVMBaseOption(String key, String description, Object defaultValue, OptionParser parser, PropertyCategory category) {
-        this.key = LLVMOptions.getOptionPrefix() + key;
-        this.description = description;
-        this.defaultValue = defaultValue;
-        this.parser = parser;
-        this.category = category;
-    }
-
-    private final String key;
-    private final String description;
-    private final Object defaultValue;
-    private final OptionParser parser;
-    private final PropertyCategory category;
-
-    @Override
-    public String getKey() {
-        return key;
-    }
-
-    @Override
-    public String getDescription() {
-        return description;
-    }
-
-    @Override
-    public Object getDefaultValue() {
-        return defaultValue;
-    }
-
-    @Override
-    public String getCategoryLabel() {
-        return category.toString();
-    }
-
-    @Override
-    public Object getValue() {
-        return parser.parse(this);
-    }
-
-    public enum PropertyCategory {
-        GENERAL,
-        DEBUG,
-        PERFORMANCE,
-        TESTS,
-        MX;
-    }
+import com.oracle.truffle.llvm.option.Constants;
+import com.oracle.truffle.llvm.option.Option;
+import com.oracle.truffle.llvm.option.OptionCategory;
+
+@OptionCategory(name = "Base Options")
+abstract class LLVMBaseOption {
+
+    @Option(commandLineName = "Debug", help = "Turns debugging on/off") //
+    protected static final Boolean DEBUG = false;
+
+    @Option(commandLineName = "Verbose", help = "Enables verbose printing of debugging information") //
+    protected static final Boolean VERBOSE = false;
+
+    @Option(commandLineName = "PrintPerformanceWarnings", help = "Prints performance warnings") //
+    protected static final Boolean PRINT_PERFORMANCE_WARNINGS = false;
+
+    @Option(commandLineName = "PerformanceWarningsAreFatal", help = "Terminates the program after a performance issue is encountered") //
+    protected static final Boolean PERFORMANCE_WARNING_ARE_FATAL = false;
+
+    @Option(commandLineName = "PrintASTs", help = "Prints the Truffle ASTs for the parsed functions") //
+    protected static final Boolean PRINT_FUNCTION_ASTS = false;
+
+    @Option(commandLineName = "PrintExecutionTime", help = "Prints the execution time for the main function of the program") //
+    protected static final Boolean PRINT_EXECUTION_TIME = false;
+
+    @Option(commandLineName = "LLVM", help = "Version of the used LLVM File Format 3.2/3.8") //
+    protected static final String LLVM_VERSION = "3.2";
+
+    @Option(commandLineName = "NodeConfiguration", help = "The node configuration (node factory) to be used in Sulong.") //
```

Comment:
Nitpicking: Some of the sentences end with `.` and others don't. This was not introduced by this refactoring, but it would be a good point to make them consistent.

Could we even enforce adding a `.` (or leaving it away) in the annotation processor or would that be an overkill?